### PR TITLE
[full-ci] Small fixes

### LINKED ISSUE #5961: OC Web showing blank screen in Chrome Browser
### Steps to reproduce
1. login to account
2. click on new design

### Expected behaviour
showing OC Web Interface

### Actual behaviour
blank screen appears

### Environment general
**Operating system**:
Server: Red Hat Enterprise Linux Server release 7.3 (Maipo)
Server: Kernel: Linux 3.10.0-957.el7.x86_64
Client: Chrome Browser: Version 84.0.4147.89 (Offizieller Build) (64-Bit)
Client: Ubuntu 20.04 LTS, Mate, Linux 5.4.0-89-generic

**Backend (ownCloud Core or Infinite Scale)**:
oC 10.8.0.4

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
LDAP

### Environment ownCloud Infinite Scale
**Version:** 10.8.0

**Updated from an older Infinite Scale or fresh install:**
oC updated from 10.6.0, Web App updated from 4.2 (afair)

**Where did you install Infinite Scale from:**

### Environment ownCloud Core
**Web server:** Apache httpd.x86_64                    2.4.6-97.el7_9.1             @rhel-7-server-rpms

**Database:** mySQL/MariaDB cluster

**PHP version:**
php.x86_64                      7.4.25-1.el7.remi            @remi-php74 

**Version:** (see ownCloud admin page)

**Updated from an older ownCloud or fresh install:**

**Where did you install ownCloud from:**
customer site

**Signing status (ownCloud 9.0 and above):**

```
Login as admin user into your ownCloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results into https://gist.github.com/ and put the link here.
```

**The content of config/config.php:**

```
Log in to the web-UI with an administrator account and click on
'admin' -> 'Generate Config Report' -> 'Download ownCloud config report'
This report includes the config.php settings, the list of activated apps
and other details in a well-sanitized form.

or 

If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your ownCloud installation folder

*ATTENTION:* Do not post your config.php file in public as is. Please use one of the above
methods whenever possible. Both, the generated reports from the web-ui and from occ config:list
consistently remove sensitive data. You still may want to review the report before sending.
If done manually then it is critical for your own privacy to diligently
remove *all* hostnames, passwords, usernames, salts and other credentials before posting.
You should assume that attackers find such information and will use them against your systems.
```

**List of activated apps:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your ownCloud installation folder.
```

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** yes/no

### Client configuration
**Browser:**

**Operating system:**

#### Web server error log
```
Insert your webserver log here
```

#### ownCloud log (data/owncloud.log)
```
Insert your ownCloud log here
```

#### Browser log
```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log 
c) ...
```

### PR DESCRIPTION
**Pull request by diocas**
Just a bunch of (I hope) fixes:

- I've made the hidden files hidden by default. This is the current behavior in oc10 and I believe should be the same in web;
- Generate sourcemap. We did this to make it easier to debug, let me know if you think this should not be pushed to production (**edit:** made it configurable)
- ~~Clean paths for drawio~~ (opened a diff PR with this fix)

Some more debatable fixes:

~~I've changed the default title of the html page from "owncloud" to "loading" (not translated). Ofc this changes to the configured title once the app loads, but if something fails or takes a long time to load, users get confused.~~ (**edit:** made the title configurable instead)

In the same spirit, I've added information that is displayed before the app loads:  ~~a loading message~~ a spinner that appears if it takes more than 1s) to load resources or an error message if something fails. When we had our performance issues, users were getting blank pages and had no idea what was happening, so a bit of feedback is positive. This is not translated, but the users only see the text on error, so it should be acceptable like this.
Fixes #5961: it doesn't improve the performance, but a least gives feedback to the user instead of a white page

### CODE REVIEW

**Comment by pascalwengerter** [comment]
@diocas could you run `yarn lint --fix` and commit the changes so we can see what CI says? :)

**Comment by ownclouders** [comment]
Results for <strong>oC10Trashbin</strong> https://drone.owncloud.com/owncloud/web/20127/38/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10Trashbin</strong> https://drone.owncloud.com/owncloud/web/20133/38/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingInternalUsers2</strong> https://drone.owncloud.com/owncloud/web/20139/58/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10Trashbin</strong> https://drone.owncloud.com/owncloud/web/20175/38/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10Trashbin</strong> https://drone.owncloud.com/owncloud/web/20224/38/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10Trashbin</strong> https://drone.owncloud.com/owncloud/web/20307/38/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10Trashbin</strong> https://drone.owncloud.com/owncloud/web/20311/38/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by diocas** [comment]
Rebases with latest and did some changes:

- When loading the app takes longer than 1s, instead of showing text, which was in english and was not translated, I show a spinner similar to `ocSpinner`.
- Made the title configurable (since you prob don't want the "loading..." title, as I had before.
- You separated `ActionMenuItem` and changed the display of icon and img to an if/else. But the else is never possible, as an icon of an app is always defined (`icon: appInfo.icon || 'check_box_outline_blank'`), so I inverted it.

@kulmann @pascalwengerter can this be merged? Some of it, at least?

File: packages/web-runtime/src/store/apps.js [hunk 1]
```diff
@@ -46,11 +46,13 @@ const mutations = {
...
       app,
       icon: extension.icon,
+      img: extension.img,
       newTab: extension.newTab || false,
       routeName: extension.routeName,
       routes: extension.routes || [],
       extension: extension.extension,
-      handler: extension.handler
+      handler: extension.handler,
+      canBeDefault: 'canBeDefault' in extension ? extension.canBeDefault : true
```
Comment (kulmann) [inline packages/web-runtime/src/store/apps.js:55]:
```suggestion
      canBeDefault: extension.canBeDefault === true
```
has the same effect, right?

**Review by kulmann — CHANGES_REQUESTED** [review]
nice, thank you! 🚀 just some small comments, otherwise good to go.

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/web/20581/9/7
`Test Suites: 1 failed, 113 passed, 114 total`

A unit test needs to be fixed.

File: packages/web-app-files/src/mixins/fileActions.js [hunk 2]
```diff
@@ -62,41 +62,50 @@ export default {
...
+            },
+            canBeDefault: editor.canBeDefault,
+            componentType: 'oc-button',
+            class: `oc-files-actions-${kebabCase(
+              this.apps.meta[editor.app].name
+            ).toLowerCase()}-trigger`
+          }
+        })
+        .sort((first, second) => {
+          // We make the default action based on the order, so put the ones who can be default first
```
[the remark this answers is not in the document]
Comment (diocas) [reply packages/web-app-files/src/mixins/fileActions.js:103]: Was this a recent fix? In the past, I had non default apps being used when I clicked a file. But maybe the issue was he fact that every app was having `canBeDeFault` as true... I'll modify the comment then.

File: packages/web-app-files/src/mixins/fileActions.js [hunk 3]
```diff
@@ -214,7 +223,8 @@ export default {
         const label = this.$gettext('Open in %{ appName }')
         return {
           name: app.name,
-          img: app.icon,
+          icon: app.icon,
```
[the remark this answers is not in the document]
Comment (diocas) [reply packages/web-app-files/src/mixins/fileActions.js:226]: For us, yes, we added the icons in ODS. But this should be unified in the app provider as well (I will check and ask for changes), as I don't think is intuitive to have them mixed.

File: packages/web-runtime/src/store/apps.js (hunk 1 above)
Comment (diocas) [reply packages/web-runtime/src/store/apps.js:55]: right!

File: packages/web-app-files/src/mixins/fileActions.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (kulmann) [reply packages/web-app-files/src/mixins/fileActions.js:103]: https://github.com/owncloud/web/pull/5970 should fix that. Could you have a look to confirm?

**Comment by diocas** [comment]
@phil-davis I've changed the test. The test assumed we can delete the icon, but this is not true if you set it from the extension configuration (and since icon always has a default value...). So I removed this part and ensure that, when setting `img` this takes precedence over `icon`.

Also applied @kulmann requests and added a changelog.

**Comment by kulmann** [comment]
@wkloucek https://drone.owncloud.com/owncloud/web/20609/9/2 states `msg="execution failed: commit from DRONE_COMMIT_BEFORE not found: object not found"`, could you take a look?

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/web/20609/9/2
That's a strange drone CI fail - nothing to do with the code of this PR!

@diocas we have made a few fixes today for intermittent annoying things in CI pipelines. Please rebase to latest and force-push and you might get more joy! If it dies again the same, then @wkloucek could look or...

**Comment by kulmann** [comment]
restarted CI, smoke tests are flaky at the moment, sorry

**Comment by ownclouders** [comment]
Results for <strong>oCISBasic</strong> https://drone.owncloud.com/owncloud/web/20619/51/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by phil-davis** [comment]
A smoke-test pipeline failed. I commented in issue https://github.com/owncloud/web/issues/6082#issuecomment-984753681
We are now trying to see why that pipeline is having trouble.
I restarted CI.

**Comment by ownclouders** [comment]
Results for <strong>oCISBasic</strong> https://drone.owncloud.com/owncloud/web/20625/51/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10Trashbin</strong> https://drone.owncloud.com/owncloud/web/20625/40/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Review by kulmann — APPROVED** [review]

**Comment by phil-davis** [comment]
I made PR #6088 on top of this to run full-ci and get the tests passing. If I can achieve that tomorrow morning, then I can push any fixes here to get this passing.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingAccept</strong> https://drone.owncloud.com/owncloud/web/20826/16/1
The following scenarios passed on retry:
- webUISharingAcceptSharesToRoot/acceptShares.feature:108

**Pull request merged**